### ISSUE TITLE
More professional feel for API reference

### ISSUE THREAD

**Issue by djnugent** [3drobotics/sphinx_3dr_theme#39]
Original posted here https://github.com/dronekit/dronekit-python/issues/310#issuecomment-139930633

The API reference page for dronekit python is hard to follow and looks very foreign to a programmer. I have stopped going there because it is easier to look at the source code then find what I am looking for on that page. 

I suggest we try to match the format of current python API documentation. This will make the experience more familiar and easy to comprehend.

Here are a few python examples:
http://docs.opencv.org/modules/highgui/doc/reading_and_writing_images_and_video.html
https://docs.python.org/2/library/logging.html

Notable characteristics:
-Header/page break for every function(i.e. distinguishable dividers)
-Bold function names
-shows how the function is called in one line
-Lists and describes parameters
-Plenty of white space!
-Class name is listed in front of function name(when applicable)

Non essentials:
-Basic color scheme. Not too crazy
-Show deprecated methods

Other examples:
-Java docs has a very clean, extensible layout. I have never seen it used with python but is a great resource for an API

**Comment by hamishwillee** [comment]
Also maybe expand out the API reference as a TOC in the sidebar.

@kaitlynhova - I agree this could be better. Do you think you could find time to improve this further?

**Comment by kaitlynhova** [comment]
Yes! Now that the dronekit docs have been around for a few months l think it'd be great to improve on the overall look and refine how we display our info. -- What do you think of this new look (via Johan) for desktop and mobile? --> 
![desktop_01](https://cloud.githubusercontent.com/assets/5474124/9998849/27869bcc-6048-11e5-9f40-275de7519ea6.png)
![desktop_02](https://cloud.githubusercontent.com/assets/5474124/9998851/27886c9a-6048-11e5-96d6-1a453de37a41.png)
![desktop_search_01](https://cloud.githubusercontent.com/assets/5474124/9998847/278482ec-6048-11e5-9a5a-7afdcf63ee33.png)
![desktop_search_02](https://cloud.githubusercontent.com/assets/5474124/9998850/27885872-6048-11e5-9970-84af69cd84ec.png)
![desktop_search_03](https://cloud.githubusercontent.com/assets/5474124/9998852/278b8bf0-6048-11e5-969c-1f84ab3b6aea.png)
![desktop_search_04](https://cloud.githubusercontent.com/assets/5474124/9998848/2785a76c-6048-11e5-9346-6e6495a54b56.png)
![desktop_search_05](https://cloud.githubusercontent.com/assets/5474124/9998855/27987f36-6048-11e5-8019-055f9c3d171d.png)
![desktop_search_06](https://cloud.githubusercontent.com/assets/5474124/9998853/2795ceb2-6048-11e5-990a-cf8931595c6f.png)
![mobile_wip_01](https://cloud.githubusercontent.com/assets/5474124/9998854/2796817c-6048-11e5-86ea-6776f4383a39.png)
![mobile_wip_02](https://cloud.githubusercontent.com/assets/5474124/9998857/279b52ba-6048-11e5-919c-6b406ce967a3.png)
![mobile_wip_03](https://cloud.githubusercontent.com/assets/5474124/9998856/27993e12-6048-11e5-8262-f87eb4f6b8ec.png)
![mobile_wip_04](https://cloud.githubusercontent.com/assets/5474124/9998858/27a0af94-6048-11e5-9719-ce0932c46147.png)

**Comment by djnugent** [comment]
That looks nice!
I think it could use bars behind header(but that is just me)

I was commenting specifically on the API Reference page which wasn't shown in those sketches.
http://python.dronekit.io/automodule.html

**Comment by kaitlynhova** [comment]
ah, gotcha -- If we update the styles on the API reference page it will update how all of the docs look, which is probably a good thing at this point.

**Comment by hamishwillee** [comment]
@kaitlynhova @djnugent Yes, this is MUCH prettier.

> If we update the styles on the API reference page it will update how all of the docs look, which is probably a good thing at this point.

Yes, it is likely to be a little better, but it probably won't address most of the points Daniel made (or add the class/function structure into the sidebar). Can we have some examples of how you think it will look in the API Reference?

**Comment by hamishwillee** [comment]
@kaitlynhova You've gone quiet on this :-) Any progress/ETA?

**Comment by kaitlynhova** [comment]
@hamishwillee : sorry for the delay! My time has been spoken for in order to get Blackbox out the door. I'll ask and see if I can get this going on Dronekit and get back to y'all.

**Comment by hamishwillee** [comment]
Thanks @kaitlynhova - that'll teach you to have other priorities :-)

**Comment by kaitlynhova** [comment]
Hey y'all --- I don't have time to knock this out right now because of other priorities (Blackbox), but I'd really like to keep this issue open for the future. These are all good ideas, but I wish that I had more time right now.

**Comment by hamishwillee** [comment]
@kaitlynhova Understood. Hope you have time to look at this again sometime soon. I'll keep nagging on a regular basis.
